lib/tasks/demo.rake

```diff
@@ -1,6 +1,41 @@
 # frozen_string_literal: true
 
 namespace :demo do
+desc 'Backup the database'
+  task backup_database: :environment do
+    raise 'This task is only for use in a development environment' unless Rails.env == 'development' || ENV['DISABLE_DATABASE_ENVIRONMENT_CHECK']
+    username = ActiveRecord::Base.configurations.configurations[1].config['username']
+    database = ActiveRecord::Base.configurations.configurations[1].config['database']
+    system "mysqldump --opt --user=#{username} #{database} > sara_database_#{Time.now.to_i}.sql"
+  end
+
+  desc 'Restore the database'
+  task restore_database: :environment do
+    # Example usage: rake demo:restore_database FILE=sara_database_1606835867.sql
+    raise 'This task is only for use in a development environment' unless Rails.env == 'development' || ENV['DISABLE_DATABASE_ENVIRONMENT_CHECK']
+    username = ActiveRecord::Base.configurations.configurations[1].config['username']
+    database = ActiveRecord::Base.configurations.configurations[1].config['database']
+    system "mysql --user=#{username} #{database} < #{ENV['FILE']}"
```

Comment:
Does this mysql command append to the existing database contents or delete and write a new database?

Reply:
Deletes and writes a new database, if you look at the contents of the backup .sql file you'll see that it drops the old table (if exists), re-creates the table and then populates it with data